src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravellerSelection.tsx

```diff
@@ -27,65 +33,138 @@ export function TravellerSelection({
+  const {t, language} = useTranslation();
+  const styles = useStyles();
+  const {open: openBottomSheet, close: closeBottomSheet} = useBottomSheet();
 
-  const userCountState = useUserCountState(selectableUserProfiles);
-  const selectableUserProfilesWithCount =
-    userCountState.userProfilesWithCount.filter((a) =>
-      selectableUserProfiles.some((b) => a.id === b.id),
-    );
-  useEffect(() => {
-    if (selectionMode == 'none' && selectableUserProfiles.length > 0) {
-      // If selectionMode is none, we should default to the first item.
-      // selectableUserProfiles should be only have one item, but this
-      // is specified by the fare product and reference data in firestore.
-      setTravellerSelection([
-        {
-          ...selectableUserProfiles[0],
-          count: 1,
-        },
-      ]);
-    } else {
-      const filteredSelection = userCountState.userProfilesWithCount.filter(
-        (u) => selectableUserProfiles.find((i) => i.id === u.id),
-      );
-      setTravellerSelection(filteredSelection);
-    }
-  }, [userCountState.userProfilesWithCount, fareProductType, selectionMode]);
+  const accessibility: AccessibilityProps = {
+    accessible: true,
+    accessibilityRole: 'button',
+    accessibilityLabel:
+      t(
+        selectionMode == 'multiple'
+          ? PurchaseOverviewTexts.travellerSelection.title_multiple
+          : PurchaseOverviewTexts.travellerSelection.title_single,
+      ) + screenReaderPause,
+    accessibilityHint: t(PurchaseOverviewTexts.travellerSelection.a11yHint),
+  };
+
+  const [selectableUserProfilesWithCount, setSelectableUserProfilesWithCount] =
+    useState<UserProfileWithCount[]>(selectableUserProfiles);
 
   useEffect(() => {
-    userCountState.updateSelectable(selectableUserProfiles);
-  }, [selectableUserProfiles]);
+    const filteredSelection = selectableUserProfilesWithCount.filter((u) =>
+      selectableUserProfiles.find((i) => i.id === u.id),
+    );
+    setTravellerSelection(filteredSelection);
+  }, [fareProductType, selectionMode, selectableUserProfilesWithCount]);
 
   if (selectionMode === 'none') {
-    return <></>;
+    return null;
   }
 
+  const totalTravellersCount = selectableUserProfilesWithCount.reduce(
+    (acc, sUPWC) => acc + sUPWC.count,
+    0,
+  );
+
+  const activeTravellerCategoryCount = selectableUserProfilesWithCount.filter(
+    (u) => u.count,
+  ).length;
+
+  const multipleTravellerCategoriesSelectedFrom =
+    activeTravellerCategoryCount > 1;
+
+  const multipleTravellersDetailsText =
+    totalTravellersCount == 0
+      ? t(PurchaseOverviewTexts.travellerSelection.no_travellers_selected)
+      : selectableUserProfilesWithCount
+          .filter((userProfile) => userProfile.count > 0)
+          .map(
+            (userProfile) =>
+              `${userProfile.count} ${getReferenceDataName(
+                userProfile,
+                language,
+              )}`,
+          )
+          .join(', ');
```

Comment:
This is a comment not ment as critique, but more of a discussion point. You are free to not agree as everybody got their style, but I like these kinda comments/discussions and feel they are healthy for a team to have :)

I am a fan of verbose variable names to ensure greater readability :+1: However, it is also important to not add too many verbose variables names, as it ends up kinda cluttering to read. For example the variable `activeTravellerCategoryCount` I think isn't worth the cognitive load it demands for the user to read this code. Also there are some duplicate filtering on user count. 

By removing this the code can be reduced to something like this:
```ts
const selectedUserProfiles = selectableUserProfilesWithCount.filter((u) => u.count);
const totalTravellersCount = selectedUserProfiles.reduce((acc, {count}) => acc + count, 0);
const multipleTravellerCategoriesSelectedFrom = selectedUserProfiles.length > 1;
const multipleTravellersDetailsText = selectedUserProfiles
  .map((u) => `${u.count} ${getReferenceDataName(u, language)}`
  .join(', ') || t(PurchaseOverviewTexts.travellerSelection.no_travellers_selected);
```

Also when looking at it, is the variable `multipleTravellerCategoriesSelectedFrom` easier to understand compared to just using `selectedUserProfiles.length > 1` inline in the jsx? I'm not sure, just food for thought.

Reply:
Nice touch with only one reduce iteration and destructuring :+1:

Regarding multipleTravellerCategoriesSelectedFrom I think it is good practice to keep it in its own variable since it is later used twice in a way where the conditions should stay in sync.